### ISSUE TITLE
Lua error: Internal error: The interpreter exited with status 127. 

### ISSUE THREAD

**Issue by BisuZero** [d8sychain/docker-mediawiki#8]
I'm getting a script error for my page when I try to use the Scribunto extension.

"Lua error: Internal error: The interpreter exited with status 127.

No further details are available."

**Comment by d8sychain** [comment]
@BisuZero 
At the moment Scribunto is one of the extensions that I have not tested yet, nor do I have any experience with Lua yet. I did do a quick search on the error and saw a few issues to where it may be due to execute file permissions.

https://www.mediawiki.org/wiki/Topic:S3mp0y6905f3lrhf
https://www.mediawiki.org/wiki/Topic:Tdf5bab8pn9qwvcz
https://stackoverflow.com/questions/17844370/something-incorrect-in-shell-path-syntax 

I will try to look into this more but due to limited time, I don't foresee that happening soon. If you find a working solution, please comment back here and share your results.

**Comment by dev-mcheung** [comment]
I hit a dead end on it since there isn't much solutions around for that status 127 error code. I have seen all 3 of those links during my own research on the issue.

What's odd is that the issue in the past for this has been lua file pathing but Special:Version for lua is showing up properly under Installed software. 

Everything is working great except for this blasted 127 error code from Scribunto. I'll keep an eye on this issue and hope it gets resolve in the future. Besides that, this is a great docker addition for unraid.

**Comment by BisuZero** [comment]
I hit a dead end on it since there isn't much solutions around for that status 127 error code. I have seen all 3 of those links during my own research on the issue.

What's odd is that the issue in the past for this has been lua file pathing but Special:Version for lua is showing up properly under Installed software. 

Everything is working great except for this blasted 127 error code from Scribunto. I'll keep an eye on this issue and hope it gets resolve in the future. Besides that, this is a great docker addition for unraid.

**Comment by d8sychain** [comment]
I just found out MediaWiki 1.34.0 was released a day ago. So I will be changing the dockerfile and bumping the `latest` tag with 1.34.0. 
https://lists.wikimedia.org/pipermail/mediawiki-announce/2019-December/000242.html
All of the new requirements for 1.34 are already in place so look for it probably tomorrow, after I test the upgrade process on my MW dockers 1st. Maybe the issue will resolve it's self. Also, MW 1.33.1 has upgraded to 1.33.2 with a few security fixes.
https://lists.wikimedia.org/pipermail/mediawiki-announce/2019-December/000241.html
If you want to stay with 1.33+ be sure to use the tag `d8sychain/mediawiki:1.33`.

Off-topic, are you using the Visual Editor and have you had any issues with it?

**Comment by BisuZero** [comment]
Zero issue with Visual Editor. I'll keep a test docker up and will upgrade after 1.34.0. It's odd that no one else has reported the Scribunto issue. 

Can you spin up a test instance of Mediawiki docker with Scribunto and then test the console using `mw.log()`?

I attached a screenshot on the gist of it if you have never use the extension. I just wanna make sure this isn't an isolated issue on my end.

![2019-12-21_19-51-14](https://user-images.githubusercontent.com/40740444/71315833-86ae8780-242b-11ea-88d5-25f4008428bb.png)

**Comment by d8sychain** [comment]
Okay, I wasn't planning on doing this any time soon, but I went "down the rabbit hole", and should have this fixed.
![image](https://user-images.githubusercontent.com/43706802/71321160-c739ef80-2483-11ea-92d5-37927e336c94.png)
![image](https://user-images.githubusercontent.com/43706802/71321265-e08f6b80-2484-11ea-8f43-61a3f3328d60.png)
The generic binary included with the Scribunto extension won't work in this docker based on Alpine. I've made changes to the docker/code but I won't be doing another commit until I finish some other stuff that I was working on. I sort of put everything else on hold to work on this issue, and it took me some time to climb back out of the "rabbit hole". I'll leave you with some instructions to fix this on your own if you don't want to wait on the next build.

* SSH into the docker container `docker exec -it name_of_docker_container /bin/bash`
* Run: `apk add --no-cache --upgrade lua`
* If your on UnRaid **/config** will most likely be **mnt/user/appdata/docker_container...**
* Create folder **lua** in **/config/logs/**
* In **/config/www/mediawiki/LocalSetting_Extensions.php** replace the config for Scribunto with:
```
## Scribunto https://www.mediawiki.org/wiki/Extension:Scribunto
	#wfLoadExtension( 'Scribunto' );
	$wgScribuntoEngineConf['luastandalone']['errorFile'] = '/config/log/lua/lua.log';
	$wgScribuntoEngineConf['luastandalone']['luaPath'] = '/usr/bin/lua';
```
Let know how it goes.

**Comment by BisuZero** [comment]
That fixed the issue completely with the extension, thank you! You can close this issue when you push out the next docker version with the fix.
